### DISCUSSION TITLE
Surface Book 3 Camera Support Feature Matrix Clarification

### DISCUSSION THREAD

**Discussion by aanish-pradhan** [linux-surface/linux-surface#1050] [Q&A]
In the [feature matrix](https://github.com/linux-surface/linux-surface/wiki/Supported-Devices-and-Features#feature-matrix), it shows that the Surface Book 3 cameras are not working. The key of the feature matrix indicates that an 'X' means that the feature is not supported and won't work. For clarification, does this mean that the cameras for the Surface Book 3 will NEVER work/be supported. I've been following the [Camera support issue](https://github.com/linux-surface/linux-surface/issues/91) and it seems that they recently made some development towards camera support. But, the issue is only tagged with the Surface Book and Surface Book 2 labels. So to reiterate the question, does this mean the Surface Book 3 cameras will never work at all?

**Comment by NP-chaonay** [comment]
if that indicator of "X" is not outdated

it means that the camera is not work

it has some clue that we dont know or lack of people or people time. OR even development is not nearly done yet.

So easy defination of "X" is "not work at this moment"

**Comment by WolfWings** [comment]
Instead of tracking issue 91, there's a much better set of charts at https://github.com/linux-surface/linux-surface/wiki/Camera-Support instead.

The Surface Book 3 and Surface Pro 7 both use the same camera sensors (OV5693 and OV8865) as almost all Surface devices, however they use the Image Processing Unit version 4 (IPU4) which... has no drivers, Intel has no intention of EVER upstreaming drivers, and the existing code to access the IPU4-based devices is atrocious according to numerous kernel devs, and Intel has never yet released firmware files for IPU4 even.

So no, it's unlikely the SB3 or SP7 will ever have camera support on Linux. IPU4 is a dead-end design and has been dropped entirely even by Intel now in favor of IPU6 as well, so no further IPU4 designs will show up either.
